Source/JavaScriptCore/jsc.cpp

```diff
@@ -1098,7 +1098,7 @@ class ShellSourceProvider final : public StringSourceProvider {
             return { };
         const char* cachePath = Options::diskCachePath();
         String filename = FileSystem::encodeForFileName(FileSystem::lastComponentOfPathIgnoringTrailingSlash(sourceOrigin().url().fileSystemPath()));
-        return FileSystem::pathByAppendingComponent(cachePath, makeString(source().hash(), '-', filename, ".bytecode-cache"));
+        return FileSystem::pathByAppendingComponent(StringView { cachePath }, makeString(source().hash(), '-', filename, ".bytecode-cache"_s));
```

Comment:
This should be `StringView::fromLatin1`. Not sure we even have that, but one of the other reasons we would like to not do this automatically is that these `const char*` are sometimes UTF-8 and converting to a `StringView` mishandles the encoding of the characters. `ASCIILiteral` deals with it nicely by making non-ASCII illegal, and should be a compile time error if we can ever figure out how to make that work.

Maybe eventually we’ll remove even the explicit constructor and only support the named function.

Reply:
Ok, I'll introduce StringView::fromLatin1 and use it in this patch. In a follow-up, I'll be able to drop the StringView(const char*) constructor.